src/compile.rs

```diff
@@ -101,11 +105,70 @@ fn compile_universal2(
     Ok(result)
 }
 
+/// We want to use `zig cc` as linker and c compiler. We want to call `python -m ziglang cc`, but
+/// cargo only accepts a path to an executable as linker, so we add a wrapper script. We then also
+/// use the wrapper script to pass arguments and substitute an unsupported argument.
+///
+/// We create different files for different args because otherwise cargo might skip recompiling even
+/// if the linker target changed
+fn prepare_zig_linker(context: &BuildContext) -> Result<PathBuf> {
+    let (zig_linker, target) = match context.platform_tag {
+        None | Some(PlatformTag::Linux) => (
+            "./zigcc-gnu.sh".to_string(),
+            "native-native-gnu".to_string(),
```

Comment:
We can also support cross compiling with zig. I just tried locally and it works if we give zig cc the right target triple.

Reply:
Yes, I've seen https://actually.fyi/posts/zig-makes-rust-cross-compilation-just-work/ and it would awesome if we could port that to maturin! Do you still have to download some built python or specific headers for the target platform to make this work, and if so, what's the easiest way to download them?

Reply:
> Do you still have to download some built python or specific headers for the target platform to make this work

Yes, it's needed for non-abi3 wheels.

> and if so, what's the easiest way to download them?

Maybe https://github.com/indygreg/python-build-standalone can be used.

Reply:
On second thought, for linux maybe we can just collect sysconfigs from manylinux docker images and use them as cross compiling target Python sysconfigdata (something like [this](https://github.com/PyO3/maturin/issues/375#issuecomment-758642486)).